File: leetcode/easy/symmetric_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def _isSymmetric(one, two):
            if(not one and not two):
                return True
            if(not one or not two):
                return False
            if(one.val != two.val):
                return False
            return _isSymmetric(one.left, two.right) and _isSymmetric(one.right, two.left)

        if(not root):
            return True
        return _isSymmetric(root.left, root.right)

File: leetcode/easy/symmetric_tree.ts
import { TreeNode } from "../common/common";

function isSymmetric(root: TreeNode | null): boolean {
  function _isSymmetric(
    first: TreeNode | null,
    second: TreeNode | null
  ): boolean {
    if (first === null && second === null) {
      return true;
    }
    if (first === null || second === null) {
      return false;
    }
    if (first.val != second.val) {
      return false;
    }
    return (
      _isSymmetric(first.left, second.right) &&
      _isSymmetric(first.right, second.left)
    );
  }

  if (root === null) return true;
  return _isSymmetric(root.left, root.right);
}
